library/std/src/os/unix/net/ancillary.rs

```diff
@@ -334,6 +407,7 @@ pub enum AncillaryData<'a> {
     ScmRights(ScmRights<'a>),
     #[cfg(any(doc, target_os = "android", target_os = "linux",))]
     ScmCredentials(ScmCredentials<'a>),
+    IpTtl(u8),
```

Comment:
Does it really make sense to mix all of these in the same type?

Maybe there should be different enums for Unix sockets and for IP sockets, since they have completely different types of ancillary data. If we mix them all together, a user must still deal with IP TTL messages when using Unix sockets.

Similarly, it would allows IP TTL data to ancillary data meant for a unix socket.

Reply:
You are right, but I thought in the future there would have some cases in common.

Reply:
Hmm, I do think it would make sense to have separate types for different sockets, even if they share a common subset of ancillary data. Maybe it should even be different for each socket type, not just per address family, since IP_TTL is specific to UDP (and probably SCTP, but we don't support that at all now).

Reply:
This would mean, that we currently have to split it in 3 different enums. I think this would lead to code duplication, or at least not so good readable code.

Reply:
I split the enum to `UnixAncillary` and `IpAncillary`.